pkg/apis/authentication/types.go

```diff
@@ -162,3 +162,24 @@ type BoundObjectReference struct {
 	// UID of the referent.
 	UID types.UID
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// SelfSubjectAttributesReview contains the user information that the kube-apiserver has about
+// the user who originated the request.
+type SelfSubjectAttributesReview struct {
```

Comment:
I missed the discussion on the Type name. I guess the format is:
`[Self]Subject{Access,Rules,Attributes}Review`

Is `Attributes` adding much value here? For example, we have `TokenReview`, not `TokenAttributesReview`. I have a slight preference for dropping `Attributes`, since the thing being reviewed is the subject itself.

@deads2k, thoughts?

Reply:
I personally like the idea.

Reply:
> @deads2k, thoughts?

No preference.  `SelfSubjectReview` reads fine to me.

Reply:
Fixed. API hs been renamed everywhere (except KEP, I will do it later).